lib/TokenManager.js

```diff
@@ -91,18 +34,36 @@ function add(sdk, tokenMgmtRef, storage, key, token) {
   }
   tokenStorage[key] = token;
   storage.setStorage(tokenStorage);
-  setRefreshTimeout(sdk, tokenMgmtRef, storage, key, token);
 }
 
 function get(storage, key) {
   var tokenStorage = storage.getStorage();
   return tokenStorage[key];
 }
 
-function remove(tokenMgmtRef, storage, key) {
-  // Clear any listener for this token
-  clearRefreshTimeout(tokenMgmtRef, key);
+function getAsync(sdk, tokenMgmtRef, storage, key) {
+  var token = get(storage, key);
+  var deferred = Q.defer();
 
+  if (token && (token.expiresAt * 1000 <= Date.now())) {
+    if (tokenMgmtRef.autoRefresh) {
+      // Token has expired, refresh the token
+      refresh(sdk, tokenMgmtRef, storage, key)
```

Comment:
We should ensure that we're only calling `refresh` once, as needed.  I.e., if multiple calls to `getAsync` come in at the same time (e.g. a react app is bootstrapping, and multiple components are all triggering a downstream call to this method), we should only make one call to `refresh` and then resolve all returned promises with the same new token value.